logrotate.c

```diff
@@ -2523,6 +2524,14 @@ int main(int argc, const char **argv)
     logSetLevel(MESS_NORMAL);
     setlocale (LC_ALL, "");
 
+    if (stat(STATEDIR, &st) == -1) {
+        mkdir(STATEDIR, 0755);
+    }
+
+    if (stat(OLDSTATEFILE, &st) == 0) {
+        unlink(OLDSTATEFILE);
```

Comment:
This introduces a TOCTOU race.  Is not it better to just call ```mkdir()```?  It fails with ```EEXISTS``` if the target already exists.